substrate/frame/broker/src/migrations/v1.rs

```diff
@@ -0,0 +1,323 @@
+		fn pre_upgrade() -> Result<Vec<u8>, sp_runtime::TryRuntimeError> {
+			ensure!(StorageVersion::get::<Pallet<T>>() == 0, "can only upgrade from version 0");
+
+			let (interlude_length, configuration_leadin_length) =
+				if let Some(config_record) = v0::Configuration::<T>::get() {
+					(config_record.interlude_length, config_record.leadin_length)
+				} else {
+					((0 as u32).into(), (0 as u32).into())
+				};
+
+			log::info!(target: TARGET, "Configuration Pre-Migration: Interlude Length {:?} Leading Length {:?} ", interlude_length, configuration_leadin_length);
+
+			let (sale_start, sale_info_leadin_length) =
+				if let Some(sale_info_record) = v0::SaleInfo::<T>::get() {
+					(sale_info_record.sale_start, sale_info_record.leadin_length)
+				} else {
+					((0 as u32).into(), (0 as u32).into())
+				};
+
+			log::info!(target: TARGET, "SaleInfo Pre-Migration: Sale Start {:?} Interlude Length {:?}  ", sale_start, sale_info_leadin_length);
+
+			Ok((interlude_length, configuration_leadin_length, sale_start, sale_info_leadin_length)
+				.encode())
+		}
+
+		fn on_runtime_upgrade() -> Weight {
+			let mut weight = T::DbWeight::get().reads(1);
+
+			if StorageVersion::get::<Pallet<T>>() != 0 {
+				log::warn!(
+					target: TARGET,
+					"skipping on_runtime_upgrade: executed on wrong storage version.\
+				Expected version 0"
+				);
+				return weight
+			}
+
+			// using a u32::MAX as sentinel value in case TryFrom fails.
+			// Ref: https://github.com/paritytech/polkadot-sdk/pull/3331#discussion_r1499014975
+
+			if let Some(config_record) = v0::Configuration::<T>::take() {
+				log::info!(target: TARGET, "migrating Configuration record");
+
+				let updated_interlude_length: RelayBlockNumberOf<T> =
+					match TryFrom::try_from(config_record.interlude_length) {
+						Ok(val) => val,
+						Err(_) => u32::MAX.into(),
+					};
+
+				let updated_leadin_length: RelayBlockNumberOf<T> =
+					match TryFrom::try_from(config_record.leadin_length) {
+						Ok(val) => val,
+						Err(_) => u32::MAX.into(),
+					};
+
+				let updated_config_record = ConfigRecord {
+					interlude_length: updated_interlude_length,
+					leadin_length: updated_leadin_length,
+					advance_notice: config_record.advance_notice,
+					region_length: config_record.region_length,
+					ideal_bulk_proportion: config_record.ideal_bulk_proportion,
+					limit_cores_offered: config_record.limit_cores_offered,
+					renewal_bump: config_record.renewal_bump,
+					contribution_timeout: config_record.contribution_timeout,
+				};
+				Configuration::<T>::put(updated_config_record);
+			}
+			weight.saturating_accrue(T::DbWeight::get().reads_writes(1, 1));
+
+			if let Some(sale_info) = v0::SaleInfo::<T>::take() {
+				log::info!(target: TARGET, "migrating SaleInfo record");
+
+				let updated_sale_start: RelayBlockNumberOf<T> =
+					match TryFrom::try_from(sale_info.sale_start) {
+						Ok(val) => val,
+						Err(_) => u32::MAX.into(),
+					};
```

Comment:
Same for this. Especially we can not just take the block number and convert this to the relay chain block number. The coretime chain got launched much later than the relay chain itself. I don't want to fuck around with the sales ;)

Reply:
> Especially we can not just take the block number and convert this to the relay chain block number.

Then how should we do it?